### PR TITLE
update dependency versions

### PR DESCRIPTION
**Pull request by danmoseley**
```
dotnet tool install --global dotnet-outdated-tool
dotnet outdated -vl Major -u
```
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/4443)

### CODE REVIEW

**Comment by joperezr** [comment]
Thanks for doing this @danmoseley, change looks good to me. However, we may want to add one thing to your script before calling outdated tool: `dotnet new nugetconfig --force`. This will make sure that the feed used for the updates is nuget.org, as that will be the only way of knowing if a dependency actually has a new version.

**Comment by danmoseley** [comment]
Yes, you're quite right, I did this for aspire-samples and forgot this one's different. Let me see.

**Comment by joperezr** [comment]
Also, I don't think we want to do version lock on Major, at least not for the Aspire Component dependencies, as we said that we would be always on latest stable, regardless of it being a major version bump.

**Comment by danmoseley** [comment]
You shared your script somewhere last time and I should have just used that...

**Comment by danmoseley** [comment]
Looks like the Azure OpenAI package has breaking changes in 2.x. Reverting that.

**Comment by joperezr** [comment]
> Looks like the Azure OpenAI package has breaking changes in 2.x. Reverting that.

Ooh, first interesting one then. I guess for this particular case it is okay because our component that references that isn't stable yet, so we just have to update and address the breaks. cc: @eerhardt @sebastienros

File: Directory.Packages.props [hunk 1]
```diff
@@ -128,24 +128,24 @@
     <PackageVersion Include="OpenTelemetry.Instrumentation.GrpcNetClient" Version="1.8.0-beta.1" />
     <PackageVersion Include="OpenTelemetry.Instrumentation.Http" Version="1.8.1" />
     <PackageVersion Include="OpenTelemetry.Instrumentation.AspNetCore" Version="1.8.1" />
-    <PackageVersion Include="OpenTelemetry.Instrumentation.Runtime" Version="1.8.0" />
+    <PackageVersion Include="OpenTelemetry.Instrumentation.Runtime" Version="1.8.1" />
     <!-- build dependencies -->
     <PackageVersion Include="MicroBuild.Plugins.SwixBuild.Dotnet" Version="1.1.87-gba258badda" />
-    <PackageVersion Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="4.9.2" />
+    <PackageVersion Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="4.10.0" />
```
Comment (joperezr) [inline Directory.Packages.props:134]: I think I ask this every time (will write a note this time for myself) but just to triple check, this isn't going to mean that in order to use Aspire you need to use a newer version of VS, right @eerhardt?

**Review by joperezr — APPROVED** [review]

**Comment by danmoseley** [comment]
opened https://github.com/dotnet/aspire/issues/4466

**Comment by danmoseley** [comment]
aaand Dependabot comes to life and opens PR's for many of these. @eerhardt if you can answer the question above (or we can snip that update) I suggest we merge this one as it will likely avoid merge conflicts. Any remaining ones we can merge after. @joperezr 

Also I think this should be the last time we do this so manually

**Comment by joperezr** [comment]
I'm pretty sure the CodeAnalysis update is okay to take too as that source generator we only use internally, but totally okay to update that later too. Nice to hear that dependabot decided to work again :)

**Comment by danmoseley** [comment]
```

Aspire.Workload.Tests.StarterTemplateRunTests.ResourcesShowUpOnDashboad [1 s]
  Error Message:
   System.AggregateException : One or more errors occurred. (
╔══════════════════════════════════════════════════════════╗
║ Host system is missing dependencies to run browsers.     ║
║ Please install them with the following command:          ║
║                                                          ║
║     sudo pwsh bin/Debug/netX/playwright.ps1 install-deps ║
║                                                          ║
║ Alternatively, use apt:                                  ║
║     sudo apt-get install libgbm1\                        ║
║         libxkbcommon0                                    ║
``` 
@radical ?

**Comment by radical** [comment]
> ```
> 
> Aspire.Workload.Tests.StarterTemplateRunTests.ResourcesShowUpOnDashboad [1 s]
>   Error Message:
>    System.AggregateException : One or more errors occurred. (
> ╔══════════════════════════════════════════════════════════╗
> ║ Host system is missing dependencies to run browsers.     ║
> ║ Please install them with the following command:          ║
> ║                                                          ║
> ║     sudo pwsh bin/Debug/netX/playwright.ps1 install-deps ║
> ║                                                          ║
> ║ Alternatively, use apt:                                  ║
> ║     sudo apt-get install libgbm1\                        ║
> ║         libxkbcommon0                                    ║
> ```
> 
> @radical ?

Ah, I see it..

```
+ sudo apt-get install -y libgbm1 libxkbcommon0
E: Could not get lock /var/lib/dpkg/lock-frontend. It is held by process 1669 (dpkg)
E: Unable to acquire the dpkg frontend lock (/var/lib/dpkg/lock-frontend), is another process using it?
```

`apt-get` failed to install the dependencies, but we don't stop the test run because of that. And so it goes on to fail later in the run. Re-run should "fix" it for this PR.

In a separate one I can make it fail immediately if `apt-get` fails. Additionally, we should have some retries around this, and I remember looking at it for `dotnet/runtime`, and I can try to add something here.

**Comment by radical** [comment]
Tracking it in https://github.com/dotnet/aspire/issues/4487 .

**Comment by danmoseley** [comment]
Thanks

**Comment by eerhardt** [comment]
> > Looks like the Azure OpenAI package has breaking changes in 2.x. Reverting that.
> 
> Ooh, first interesting one then. I guess for this particular case it is okay because our component that references that isn't stable yet, so we just have to update and address the breaks. cc: @eerhardt @sebastienros

This is known. The Azure SDK team rewrote the OpenAI library. See https://devblogs.microsoft.com/dotnet/openai-dotnet-library/.

Thanks for opening https://github.com/dotnet/aspire/issues/4466 to track this. I've also opened https://github.com/dotnet/aspire/issues/4567 to track adding a plain "OpenAI" component given this new library.

File: Directory.Packages.props (hunk 1 above)
Comment (eerhardt) [reply Directory.Packages.props]: > this isn't going to mean that in order to use Aspire you need to use a newer version of VS

That is correct. Notice the comment above this: `<!-- build dependencies -->`. We only use this version in our JSON schema generator.

**Pull request merged**